snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,263 @@
+        #                              the version on pypi.
+        # --dest: Download packages into the directory we've set aside for it.
+        self._run(['download', '--disable-pip-version-check', '--dest',
+                   self._python_package_dir] + list(args), **kwargs)
+
+    def install(self, args, **kwargs):
+        """Using the requested args, install packages from the cache.
+
+        Simple wrapper for `pip install`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        the packages should have already been downloaded via `self.download`.
+
+        :param list args: Arguments for `pip install`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+        """
+        if not args:
+            return  # No packages to install
+
+        # Using pip with a few special parameters:
+        #
+        # --user: Install packages to PYTHONUSERBASE, which we've pointed to
+        #         the installdir.
+        # --no-compile: Don't compile .pyc files. FIXME: Not sure why
+        # --no-index: Don't hit pypi, assume the packages are already
+        #             downloaded (i.e. by using `self.download()`)
+        # --find-links: Provide the directory into which the packages should
+        #               have already been fetched
+        # --ignore-installed: If the package is already installed, reinstall
+        #                     it. FIXME: Not sure why
+        self._run(['install', '--user', '--no-compile', '--no-index',
+                   '--find-links', self._python_package_dir,
+                   '--ignore-installed'] + list(args), **kwargs)
+
+    def wheel(self, args, **kwargs):
+        """Using the requested args, build wheels of packages in the cache.
+
+        Simple wrapper for `pip wheel`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        the packages should have already been downloaded via `self.download`.
+
+        :param list args: Arguments for `pip wheel`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+
+        :return: List of paths to each wheel that was built.
+        :rtype: list
+        """
+        if not args:
+            return  # No wheels to build
+
+        wheels = []
+        with tempfile.TemporaryDirectory() as temp_dir:
+
+            # Using pip with a few special parameters:
+            #
+            # --no-index: Don't hit pypi, assume the packages are already
+            #             downloaded (i.e. by using `self.download()`)
+            # --find-links: Provide the directory into which the packages
+            #               should have already been fetched
+            # --wheel-dir: Build wheels into a temporary working area rather
+            #              rather than cwd. We'll copy them over. FIXME: Why
```

Comment:
to have a central "across steps" repo, for which install would prefer the prebuilt wheel, this is debatable though. If you see benefits to not doing this and nothing breaks I am all for it.

Take into account that this builds wheels for the dependencies and not just what is pointed by `source`.

Reply:
I updated the FIXME to indicate that we should try building in the package_dir once the refactor has been validated.